gluten-data/src/main/scala/org/apache/spark/sql/execution/GlutenColumnarToRowExec.scala

```diff
@@ -49,17 +49,18 @@ case class GlutenColumnarToRowExec(child: SparkPlan)
     val schema = child.schema
     for (field <- schema.fields) {
       field.dataType match {
-        case d: BooleanType =>
-        case d: ByteType =>
-        case d: ShortType =>
-        case d: IntegerType =>
-        case d: LongType =>
-        case d: FloatType =>
-        case d: DoubleType =>
-        case d: StringType =>
-        case d: TimestampType =>
-        case d: DateType =>
-        case d: BinaryType =>
+        case _: BooleanType =>
+        case _: ByteType =>
+        case _: ShortType =>
+        case _: IntegerType =>
+        case _: LongType =>
+        case _: FloatType =>
+        case _: DoubleType =>
+        case _: StringType =>
+        case _: TimestampType =>
+        case _: DateType =>
+        case _: BinaryType =>
+        case _: DecimalType =>
```

Comment:
This is a scala operator used by both native VeloxToRowConverter and ArrowToRowConverter, but the supported types are different for the two converters. Maybe we can find a way to call the native validation to ensure the scala validation is correct.
https://github.com/oap-project/gluten/blob/main/cpp/velox/compute/VeloxBackend.cc#L392-L409

Reply:
Thanks for your insightful comment! Just make separate checking in scala code to reflect the actual support state in `VeloxToRowConverter` or `ArrowColumnarToRowConverter`.